app/pages/auth_redirect_plugin/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect } from 'react';
+import { useRouter } from 'next/router';
+import { Card, Layout, Spin } from 'antd';
+import { authorizeWithAccessCode } from '../../lib/api/auth';
+
+const AuthRedirectPlugin = (): JSX.Element => {
+    const router = useRouter();
+    const { code, state } = router.query;
+
+    // TODO: move this into the AuthContext, and potentially merge this with auth_redirect (keeping both routes though)
+    useEffect(() => {
+        const authorize = async () => {
+            if (
+                code !== undefined && typeof code === 'string'
+                && state !== undefined && typeof state === 'string'
+            ) {
+                try {
+                    // TODO: handle this response - this should be handled like the normal UI login
+                    // it responds with a normal UI session (so the user doesn't have to log in again in the future)
+                    await authorizeWithAccessCode(code, state);
+                } catch (e) {
+                    console.log(e);
+                }
+            }
+        };
+
+        authorize();
+    }, [code, state]);
```

Comment:
If we are adding a different page for plugin auth, I think we should try to combine the code of the different pages (normal auth and plugin auth) with this PR. Otherwise we will probably just leave it hanging around since it works. We can at least extract this `authorize` method so that it can be used in both the normal and plugin auth pages. I think that should do the trick for generalizing the auth process.